web-client/src/ustc-ui/Pagination/Paginator.tsx

```diff
@@ -1,51 +1,74 @@
+import { Button } from '@web-client/ustc-ui/Button/Button';
 import React from 'react';
-import ReactPaginate from 'react-paginate';
+import classNames from 'classnames';
 
 export const Paginator = ({
-  breakClassName,
-  forcePage,
-  id,
-  marginPagesDisplayed = 3,
+  currentPageIndex,
   onPageChange,
-  pageCount,
-  pageRangeDisplayed,
+  totalPages,
 }: {
-  breakClassName?: string;
-  marginPagesDisplayed?: number;
-  pageCount: number;
-  id?: string;
-  pageRangeDisplayed: number;
-  onPageChange: (selectedItem: { selected: number }) => void;
-  forcePage: number;
+  totalPages: number;
+  currentPageIndex: number;
+  onPageChange: (currentPage: number) => void;
 }) => {
+  let currentPage = currentPageIndex + 1;
+
+  const nextDisabled = currentPage >= totalPages;
+  const previousDisabled = currentPage <= 1;
+
   return (
-    <>
-      <nav
-        aria-label="Pagination"
-        className="usa-pagination margin-bottom-0 margin-top-0"
-        id={id}
-      >
-        <ReactPaginate
-          activeLinkClassName="paginator-current"
-          breakClassName={`usa-pagination__item ${breakClassName}`}
-          breakLinkClassName="usa-pagination__button cursor-pointer border-none"
-          className="usa-pagination__list"
-          disabledClassName="hide"
-          forcePage={forcePage}
-          marginPagesDisplayed={marginPagesDisplayed}
-          nextClassName="usa-pagination__item"
-          nextLinkClassName="usa-pagination__link usa-pagination__next-page cursor-pointer"
-          pageClassName="usa-pagination__item usa-pagination__page-no"
-          pageCount={pageCount}
-          pageLinkClassName="usa-pagination__button cursor-pointer"
-          pageRangeDisplayed={pageRangeDisplayed}
-          previousClassName="usa-pagination__item"
-          previousLinkClassName="usa-pagination__link usa-pagination__previous-page cursor-pointer"
-          renderOnZeroPageCount={null}
-          onPageChange={onPageChange}
-        />
-      </nav>
-    </>
+    <nav
+      aria-label="Pagination"
+      className="usa-pagination margin-bottom-0 margin-top-0"
+      role="navigation"
+    >
+      <ul className="usa-pagination__list">
+        <li className="usa-pagination__item">
+          <Button
+            link
+            aria-disabled={previousDisabled}
+            aria-label="Previous"
+            className={classNames(
+              `${previousDisabled && 'hide'}`,
```

Comment:
Funnily enough, looking at `hide` (which just does `display: none`), I'm not sure it is any better than conditional rendering. I'm not an expert on accessibility by any means, but I think we might want to use [the sr-only class](https://github.com/ustaxcourt/ef-cms/blob/97ec44dfb6354a0e5c619b0181fd8b57e2bdd25d/web-client/src/styles/custom.scss#L2326C1-L2336C2) that I added for a related ticket. See, e.g., https://piccalil.li/blog/visually-hidden/ and https://stackoverflow.com/questions/19758598/what-is-sr-only-in-bootstrap-3.

In any case, I think the code is at least better-structured this way, and I'll let you decide if you want to use `sr-only` or not.

(NB: It looks like we use the same `display: none` approach for `usa-sr-only`, which is probably incorrect and which we will probably want to change at some point. But that's outside the scope of this ticket.)

Reply:
Let's fix this!